includes/discovery/functions.inc.php

```diff
@@ -953,3 +953,35 @@ function ignore_storage($descr)
 
     return $deny;
 }
+
+/**
+ * @param $types
+ * @param $device
+ * @param array $pre_cache
+ */
+function sensors($types, $device, $pre_cache = array())
+{
+    global $config;
+    if (is_array($types)) {
```

Comment:
You can avoid the is_array check by casting $types to array `foreach ((array)$types as $type)`